ccmlib/node.py

```diff
@@ -974,20 +974,19 @@ def clear(self, clear_all=False, only_data=False, saved_caches=False):
         self._create_directory()
 
     def run_sstable2json(self, out_file=None, keyspace=None, datafiles=None, column_families=None, enumerate_keys=False):
-        print("running")
         if out_file is None:
-            out_file = sys.stdout
+            out_file = subprocess.PIPE
         sstable2json = self._find_cmd('sstabledump')
         env = self.get_env()
         sstablefiles = self.__gather_sstables(datafiles, keyspace, column_families)
-        print(sstablefiles)
+        common.print_if_standalone(str(sstablefiles), debug_callback=self.info, end='')
         for sstablefile in sstablefiles:
-            print(f"-- {os.path.basename(sstablefile)} -----")
+            common.print_if_standalone(f"-- {os.path.basename(sstablefile)} -----", debug_callback=self.info, end='')
             args = sstable2json + [sstablefile]
             if enumerate_keys:
                 args = args + ["-e"]
-            subprocess.call(args, env=env, stdout=out_file)
-            print("")
+            res = subprocess.run(args, env=env, stdout=out_file, text=True, check=True)
+            common.print_if_standalone('' if out_file else res.stdout, debug_callback=self.info, end='')
```

Comment:
i think `out_file` always evaluates to `True` here. as we assign `subprocess.PIPE` to it couple lines above. assuming we don't pass a `out_file` whose `__bool__()` returns `False`. so probably we can do something like:
```py
if res.stdout is None:
    common.print_if_standalone('', debug_callback=self.info, end='')
else:
    common.print_if_standalone(res.stdout, debug_callback=self.info, end='')
```

Reply:
you are correct, change the logic to look at `res.stdout` as you suggested